python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -1221,6 +1221,21 @@ def test_array_from_pandas_typed_array_with_mask(self, t, data, expected):
         assert pa.Array.from_pandas(expected,
                                     type=pa.list_(t())).equals(result)
 
+    def test_table_column_subset_metadata(self):
```

Comment:
Can you add 2 tests: 1 against a datetime index and 1 against another index that isn't datetime? We need to make sure the `i >= ncolumns` condition is tested.

You'll need to pass `preserve_index=True` to `pa.Table.from_pandas()`.

Reply:
Yes, I was actually playing with such frames interactively testing it out, but forgot to add to the actual tests. 

`preserve_index=True` is the default, but I can add it for explicitness if you want